### ISSUE TITLE
not working with embedded player on lazygoat.tv

### ISSUE THREAD

**Issue by garbb** [younesaassila/ttv-lol-pro#262]
**Describe the bug**
When viewing a channel on twitch.tv with TTV LOL pro enabled pre-roll ads and "preparing your stream" pre-roll message are not displayed, however when viewing the same twitch channel in embedded player on lazygoat.tv or simultwitch.com, "preparing your stream" pre-roll message is displayed.

When using plain "player.twitch.tv" address by itself in a tab, "preparing your stream" pre-roll message is not displayed.

**To Reproduce**
Unfortunately ads may or not be displayed depending on your location, however this works for me:
- Enable TTV LOL Pro
- visit "https://www.twitch.tv/imnotrightnetwork", pre-roll ads and "preparing your stream" pre-roll message are not displayed.
- visit "https://lazygoat.tv/collection/CWbLxmzRBzmUbKGXi5kMbx" (embed of same twitch channel), "preparing your stream" pre-roll message **IS** displayed.
- visit "https://simultwitch.com/imnotrightnetwork" (embed of same twitch channel), "preparing your stream" pre-roll message **IS** displayed.
- visit "https://player.twitch.tv/?channel=imnotrightnetwork&parent=lazygoat.tv", pre-roll ads and "preparing your stream" pre-roll message are not displayed.

**Expected behavior**
I expect the pre-roll message to not appear on lazygoat.tv or simultwitch.com with the embedded twitch player.

**Screenshots**
![image](https://github.com/younesaassila/ttv-lol-pro/assets/4453789/3ae92afd-9ab7-4d9c-b97f-b9c870d29af9)

**Desktop (please complete the following information):**

- OS: windows 10
- Browser: brave
- Version: 1.58.137 (chromium 117.0.5938.153)

**Additional context**
I found some other old issues about support being added for embedded players with the "player.twitch.tv" host, however this still doesn't seem to be working on lazygoat.tv or simultwitch.com?

**Comment by zGato** [comment]
Twitch streams embedded into a web page always serves that ad, even if you don't have the adblocker enabled.

There's nothing that can be done here, as it's on Twitch's end and the developer of the website using their embedded stream API.

**Comment by garbb** [comment]
What I don't understand is that on that page that I have screencapped it appears that the embedded player is in an iframe?:
![image](https://github.com/younesaassila/ttv-lol-pro/assets/4453789/11816eaf-04b0-4c62-bc68-f3fdf105579b)
with src="https://player.twitch.tv?channel=imnotrightnetwork&controls=false&height=100%25&parent=simultwitch.com&referrer=https%3A%2F%2Fsimultwitch.com%2Fimnotrightnetwork&width=100%25/"
So shouldn't this iframe behave the same as a separate tab that loads the url "https://player.twitch.tv?channel=imnotrightnetwork&controls=false&height=100%25&parent=simultwitch.com&referrer=https%3A%2F%2Fsimultwitch.com%2Fimnotrightnetwork&width=100%25/" directly (with regards to how TTV  LOL PRO interacts with it)?

**Comment by garbb** [comment]
I guess you are right?
I did some more testing:
I hosted my own local web server and loaded an html page with just
http://127.0.0.1/iframe_test.htm
```
<iframe src="https://player.twitch.tv?channel=imnotrightnetwork&parent=127.0.0.1"></iframe>
```
And it always gets that "preparing stream" pre-roll.

Whereas if I just go to the url "https://player.twitch.tv?channel=imnotrightnetwork&parent=127.0.0.1" it doesn't get the pre-roll.
Seems like there should be some workaround for this? I assume the browser must be telling player.twitch.tv somehow in a request that it is being loaded into an iframe? Is there any way to make the browser not tell the host that the url is being loaded into an iframe?
(i have a feeling that this has something to do with cross-stite scripting security or something like that?)
